idaes/models_extra/temperature_swing_adsorption/fixed_bed_tsa0d.py

```diff
@@ -0,0 +1,4522 @@
+                else:
+                    _log.warning(
+                        "Initialization of " + str(cycle_step).split(".")[-1] +
+                        " step: step 1a - iteration {0}, Failed {1}."
+                        .format(iter, cycle_step.name))
+
+                # save solution in new initial guess, f(x_new)
+                if str(cycle_step).split(".")[-1] == "heating":
+                    f_x_new = value(
+                        self.temperature_desorption -
+                        cycle_step.temperature[1])
+                else:
+                    f_x_new = value(
+                        cycle_step.temperature[1] -
+                        self.temperature_adsorption)
+
+                # set up new interval until find one containing the root
+                if f_x_new >= 0:
+                    x1 = x_new
+                    f_x1 = f_x_new
+                    condition = False
+                else:
+                    x0 = x_new
+                    f_x0 = f_x_new
+
+        # implementing false position method
+        init_log.info_high(
+            "Initialization of " + str(cycle_step).split(".")[-1] +
+            " step: step 1b: implementing false position method")
+
+        # counter
+        iter = 1
+        condition = True
+
+        # check condition to stop
+        while condition:
+
+            # compute new approximated root as x2
+            x2 = x0 - (x1 - x0) * f_x0 / (f_x1 - f_x0)
+
+            # fix time to x_2 guess
+            cycle_step.time.fix(x2)
+
+            # solve model
+            with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+                res = opt.solve(cycle_step, tee=slc.tee)
+
+            if check_optimal_termination(res):
+                init_log.info_high(
+                    "Initialization of " + str(cycle_step).split(".")[-1] +
+                    " step: step 1b - iteration {0}, completed {1}."
+                    .format(iter, idaeslog.condition(res)))
+            else:
+                _log.warning(
+                    "Initialization of " + str(cycle_step).split(".")[-1] +
+                    " step: step 1b - iteration {0}, Failed {1}."
+                    .format(iter, cycle_step.name))
+
+            # save solution in new x_2, f(x_2)
+            if str(cycle_step).split(".")[-1] == "heating":
+                f_x2 = value(
+                    self.temperature_desorption -
+                    cycle_step.temperature[1])
+            else:
+                f_x2 = value(
+                    cycle_step.temperature[1] -
+                    self.temperature_adsorption)
+
+            # check if f(x_0)*f(x_2) is negative
+            if f_x0 * f_x2 < 0:
+                x1 = x2
+            else:
+                x0 = x2
+
+            # update counter and set up new condition |f(x_2)| > error
+            iter += 1
+            condition = (f_x2**2)**0.5 > 1
+
+    def initialize_build(blk,
```

Comment:
We are starting to move to a new API for initialization, so I would recommend we update this to use the new API so that we don;t have to come back later.

Reply:
Added support for the new initialization API